runtime/drivers/duckdb/olap.go

```diff
@@ -116,6 +116,12 @@ func (c *connection) Execute(ctx context.Context, stmt *drivers.Statement) (res
 	// NOTE: We can't just "defer release()" because release() will block until rows.Close() is called.
 	// We must be careful to make sure release() is called on all code paths.
 
+	var cancelFunc context.CancelFunc
+	if stmt.ExecutionTimeout != 0 {
+		ctx, cancelFunc = context.WithTimeout(ctx, stmt.ExecutionTimeout)
+		_ = cancelFunc
```

Comment:
If not using `defer`, the `cancelFunc` must manually be called on all the error conditions